infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/config/LogReplicationConfig.java

```diff
@@ -103,6 +103,6 @@ public LogReplicationConfig(@NonNull LogReplicationSession session,
             this.maxCacheSize = serverContext.getLogReplicationCacheMaxSize();
             this.maxSnapshotEntriesApplied = serverContext.getMaxSnapshotEntriesApplied();
         }
-        this.maxDataSizePerMsg = maxMsgSize * DATA_FRACTION_PER_MSG / 100;
+        this.maxDataSizePerMsg = Math.min(maxMsgSize, serverContext.getMaxWriteSize() * DATA_FRACTION_PER_MSG / 100);
```

Comment:
I think we should have a check to make sure we are atleast 80% of the default maxUncompressedWriteSize. Just so we minimize possibility of going over